docker_custodian/docker_gc.py

```diff
@@ -46,16 +48,21 @@ def is_container_old(container, min_date):
     if state.get('Ghost'):
         return True
 
+    created_date = dateutil.parser.parse(container['Created'])
+
+    # Don't delete recently created containers
+    if max_date and created_date > max_date:
+        return False
+
     # Container was created, but never used
-    if (state.get('FinishedAt') == YEAR_ZERO and
-            dateutil.parser.parse(container['Created']) < min_date):
+    if state.get('FinishedAt') == YEAR_ZERO and created_date < min_date:
```

Comment:
Something is slipping through the cracks with the current logic. The YEAR_ZERO stuff is surprising and fragile IMO (I wouldn't be surprised if this changed in the future). A general "don't touch a container if it was created in the last <period of time> regardless of what state it is in" type of rule seems simple and effective. I'd actually advocate removing the YEAR_ZERO stuff but that is for another day.